File: src/io/innofang/AbstractFactory/example/MangoSquareCake.java
package io.innofang.AbstractFactory.example;

/**
 * Created by InnF on 2017/2/25.
 * 芒果方形蛋糕
 */
public class MangoSquareCake extends CakeFactory {
    @Override
    public CakeCream cream() {
        return new MangoCream();
    }

    @Override
    public CakeStyle style() {
        return new SquareStyle();
    }
}

File: src/io/innofang/abstract_factory/example/kotlin/MangoSquareCake.kt
package io.innofang.abstract_factory.example.kotlin

/**
 * Created by Inno Fang on 2017/9/2.
 */
class MangoSquareCake: CakeFactory() {
    override fun cream(): CakeCream {
        return MangoCream()
    }

    override fun style(): CakeStyle {
        return SquareStyle()
    }
}